test/runnable/testcontracts.d

```diff
@@ -402,6 +402,90 @@ void test7218()
 }
 
 /*******************************************/
+// 7517
+
+void test7517()
+{
+    static string result;
+
+    interface I
+    {
+        static I self;
+
+        void setEnable()
+        in
+        {
+            assert(self is this);
+            result ~= "I.setEnable.in/";
+            assert(!enabled);
+        }
+        out
+        {
+            assert(self is this);
+            result ~= "I.setEnable.out/";
+            assert( enabled);
+        }
+
+        void setDisable()
+        in
+        {
+            assert(self is this);
+            result ~= "I.setDisable.in/";
+            assert( enabled);
+        }
+        out
+        {
+            assert(self is this);
+            result ~= "I.setDisable.out/";
+            assert(!enabled);
+        }
+
+        @property bool enabled() const;
+    }
+
+    class C : I
+    {
+        static C self;
+
+        void setEnable()
+        in {}       // supply in-contract to invoke I.setEnable.in
```

Comment:
@nemanja-boric-sociomantic since it's a breaking change, I think that we should start a forum discussion.
My hope is that since the behavior proposed by your PR is more intuitive, it will get supported by enough people to be accepted.